docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx

```diff
@@ -133,63 +120,306 @@ In the following table, we compare the supported methods for managing and passin
+
+### Using agent configuration (Hybrid)
+
+<Note>Only supported for Hybrid deployments.</Note>
+
+For Dagster Cloud Hybrid deployments, making environment variables accessible is accomplished by adding them to your agent's configuration. How this is accomplished depends on the agent type.
+
+**Note**: The methods described below will make the environment variables available for all code locations in your Dagster Cloud full deployment. To configure environment variables on a code location basis, refer to the [configuration reference for your agent type](/dagster-cloud/deployment/agents).
+
+Click the tab for your agent type to view instructions.
+
 <TabGroup>
-  <TabItem name="GitHub Action Secrets (Serverless)">
-<Note>Only supported for Serverless deployments.</Note>
+  <TabItem name="Amazon ECS">
+To make environment variables accessible to a full deployment with an Amazon ECS agent, you'll need to modify the agent's CloudFormation template as follows:
 
-**Step 1: Create GitHub Action Secrets**
+1. Sign in to your AWS account.
 
-<AddGitubRepositorySecret />
+2. Navigate to **CloudFormation** and open the stack for the agent.
 
-**Step 2: Uncomment the env_vars property**
+3. Click **Update**.
 
-To make the secrets you created available, you'll need to modify two files in your GitHub repository.
+4. Click **Edit template in designer**.
 
-In the `.github/workflows/branch_deployments.yml` and `.github/workflows/deploy.yml` files, uncomment the `env_vars` property as follows:
+5. In the section that displays, click **View in Designer**. The AWS template designer will display.
 
-```yaml
-# Uncomment to pass through Github Action secrets as a JSON string of key-value pairs
-env_vars: ${{ toJson(secrets) }}
-```
+6. In the section displaying the template YAML, locate the `AgentTaskDefinition` section:
+
+   <Image
+   alt="Highlighted AgentTaskDefinition section of the AWS ECS agent CloudFormation template in the AWS Console"
+   src="/images/dagster-cloud/developing-testing/environment-variables/aws-ecs-cloudformation-template.png"
+   width={1566}
+   height={1232}
+   />
+
+7. In the `user_code_launcher.config` portion of the `AgentTaskDefinition` section, add the environment variables as follows:
 
-The secrets you created in the repository will now be available to your Dagster code as environment variables.
+   ```yaml
+   user_code_launcher:
+     module: dagster_cloud.workspace.ecs
+     class: EcsUserCodeLauncher
+     config:
+       cluster: ${ConfigCluster}
+       subnets: [${ConfigSubnet}]
+       service_discovery_namespace_id: ${ServiceDiscoveryNamespace}
+       execution_role_arn: ${TaskExecutionRole.Arn}
+       task_role_arn: ${AgentRole}
+       log_group: ${AgentLogGroup}
+       env_vars:
+         - SNOWFLAKE_USERNAME=dev
+         - SNOWFLAKE_PASSWORD       ## pulled from agent environment
```

Comment:
another (possibly better?) way of doing this would be to use the secretsmanager integration here.

There is a secrets and secrets_tag field here (just like in the run launcher in OSS: https://docs.dagster.io/deployment/guides/aws#secrets-management-in-ecs) that works in the same way.

Frustratingly these fields are not currently in the ECS agent configuration reference

The reason I say better is that you don't have to put the value here, and if you use the tag you can add and remove them without needing to add them here. But that makes an already long section even longer.

Reply:
I'm gonna punt on this for a follow up PR, but it can definitely be added in